generate-err-logs.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+output_file="deployment-err.log"
+> $output_file
+
+namespaces=$(kubectl get namespaces -o=jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}')
+
+for namespace in $namespaces; do
+    events=$(kubectl get events -n $namespace --field-selector type=Warning -o jsonpath='{range .items[*]}{.involvedObject.kind}{" "}{.involvedObject.name}{" "}{.message}{" ("}{.reason}{")\n"}{end}')
+
+    echo "$events" | while read -r kind name message; do
+        if [ "$kind" == "Pod" ]; then
+            pod=$name
+            pod_logs=$(kubectl logs -n $namespace $pod --all-containers=true 2>&1)
+            pod_definition=$(kubectl get pod -n $namespace $pod -o yaml 2>&1)
+            cat <<EOF | tee -a $output_file
+===== Logs for pod: $pod in namespace: $namespace =====
```

Comment:
I think it would be better to generate 3 files instead of putting 3 different logs in the same file. For example, syntax highlighting will work for the yaml files.

Reply:
Added, now there are 3 files:
`failed-pods-logs.log` a log containing the containers' logs of each pod
`failed-pods-definitions.yaml`  a yaml file containing the definitions of the failed logs
`failed-deployment-event-log.log` a log containing the error\warning events of failed pods